### PR TITLE
fix panic when stop

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
When stopping TiKV, the `scheduler` thread may be closed before `raftstore` thread, at that point if `raftstore` thread send msg to scheduler will cause panic.
ref https://github.com/pingcap/tikv/issues/1363
@BusyJay @disksing PTAL

### CODE REVIEW

File: src/util/transport.rs [hunk 1]
```diff
@@ -28,6 +28,10 @@ quick_error! {
             description("message is discarded")
             display("{}", reason)
         }
+        Closed {
+            description("ch is closed")
```
Comment (siddontang) [inline src/util/transport.rs]: channel is closed

File: src/storage/txn/scheduler.rs [hunk 2]
```diff
@@ -197,15 +197,19 @@ impl Drop for RunningCtx {
...
             cid: cid,
             pr: pr,
             cb_ctx: cb_ctx,
             result: result,
         }) {
-            panic!("send write finished to scheduler failed cid={}, err:{:?}",
-                   cid,
-                   e);
+            Ok(_) => {}
+            e @ Err(TransportError::Closed) => info!("ch closed, server is closing, err {:?}", e),
```
Comment (siddontang) [inline src/storage/txn/scheduler.rs]: duplicated error message here.

**Comment by siddontang** [comment]
> if raftstore thread send msg to scheduler will cause panic

can we handle the error directly, not panic?

**Comment by zhangjinpeng1987** [comment]
Must panic when send msg failed, or will cause logic error.

File: src/storage/txn/scheduler.rs [hunk 3]
```diff
@@ -678,12 +682,16 @@ impl Scheduler {
...
-            if let Err(e) = ch.send(Msg::SnapshotFinished {
+            match ch.send(Msg::SnapshotFinished {
                 cid: cid,
                 cb_ctx: cb_ctx,
                 snapshot: snapshot,
             }) {
-                panic!("send SnapshotFinish failed cmd id {}, err {:?}", cid, e);
+                Ok(_) => {}
+                e @ Err(TransportError::Closed) => {
+                    info!("ch closed, server is closing, err {:?}", e)
```
Comment (siddontang) [inline src/storage/txn/scheduler.rs]: duplicated error message

**Comment by siddontang** [comment]
PTAL @hhkbp2

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**